hometask7

```diff
@@ -0,0 +1,109 @@
+# Task 1
+import json
+
+with open('json_phonebook.json') as file:
+    phonebook = json.load(file)
+
+
+    name = input('Enter a name:')
+
+    number = input('Enter a number:')
+
+
+    if phonebook.get(name):
+        print(f'{name} already exists. You need to delete a record and add again if you want to change it')
+    else:
+        phonebook[name] = number
+
+    while True:
+        user_input = input('Enter a command:')
+        split_input = user_input.split()
+
+        command = split_input[0]
+        if command == 'stats':
+            print(len(phonebook))
+
+
+        elif command == 'add':
+            name = input('Enter a name:')
+            if not name in phonebook.keys():
+                number = input('Enter a number:')
+                phonebook[name] = number
+                print('New contact was added')
+            else:
+                name in phonebook.keys()
+                print("Fault. There is such name in a phonebook.")
+                with open('json_phonebook.json', 'a') as file:
+                    json.dump(phonebook, file)
+
+
+
+        elif command == 'del':
+            key = input('Enter some name: ')
+            if phonebook.get(key):
+                del phonebook[key]
+                print("Record was deleted")
+            else:
+                print('Such record never existed')
+                with open('json_phonebook.json', 'a') as file:
+                    json.dump(phonebook, file)
+
+
+        elif command == 'list':
+            list = (phonebook.items())
+            print(list)
+        elif command == 'show':
+            print(phonebook)
+            name = input('Enter a name:')
+            if name not in phonebook.keys():
+                print("Fault. There is no such name in a phonebook.")
+            else:
+                print(phonebook[name])
+        elif command == 'end':
+            break
+
+with open('json_phonebook.json', 'w') as file:
+    json.dump(phonebook, file)
+
+
+Task 2
+
+import time
+import os
+
+f = open('log1.txt', "w+")
```

Comment:
It is better to use `open` as a context manager. And open the file in the place where you work with it - inside the decorator.